### PR TITLE
JT-36

### PR DESCRIPTION
**Pull request by aalexeen**

### CODE REVIEW

File: stock/src/main/java/saffchen/util/validation/ValidationUtil.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * @author alex_jd on 8/10/22
+ * @project JRM-Java-Stock-System
+ */
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.NestedExceptionUtils;
+import org.springframework.lang.NonNull;
```
Comment (fstk1337) [inline stock/src/main/java/saffchen/util/validation/ValidationUtil.java:11]: Seems like unused imports.

File: stock/src/main/java/saffchen/entities/SatelliteEntity.java [hunk 2]
```diff
@@ -1,9 +1,11 @@
 package saffchen.entities;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
```
Comment (fstk1337) [inline stock/src/main/java/saffchen/entities/SatelliteEntity.java:3]: Seems to be unused.

File: stock/src/main/java/saffchen/repository/BaseRepository.java [hunk 3]
```diff
@@ -18,4 +20,8 @@
     @Modifying
     @Query("DELETE FROM #{#entityName} u WHERE u.id=:id")
     int delete(@Param("id") Long id);
+
+    default void deleteExisted(Long id) {
```
Comment (pelmegov) [inline stock/src/main/java/saffchen/repository/BaseRepository.java:24]: Why do we need this method?

**Review by pelmegov — APPROVED** [review]

**Review by fstk1337 — COMMENTED** [review]
When testing satellites endpoints, I met some issues on my machine:
1. When trying to create a new satellite (POST request) if the name of satellite already exists in database I've got HTTP 500 code error.
2. PUT method requests are not allowed by registry.
3. When I try to update a satellite sending by URL non-existing id, then application creates new satellite in database with no errors. I am not sure that such behaviour is expected.
4. When I try to update a satellite and send the satellite name that already exists in database I've got HTTP 500 code error.
5. When trying to delete a satellite with non-existing id, got 422 HTTP code error (maybe it's ok), but got no message from IllegalRequestDataException (like Entity with id={id} not found").

File: stock/src/main/java/saffchen/util/validation/ValidationUtil.java (hunk 1 above)
Comment (aalexeen) [reply stock/src/main/java/saffchen/util/validation/ValidationUtil.java:11]: removed it.

File: stock/src/main/java/saffchen/entities/SatelliteEntity.java (hunk 2 above)
Comment (aalexeen) [reply stock/src/main/java/saffchen/entities/SatelliteEntity.java:3]: Removed it.

File: stock/src/main/java/saffchen/repository/BaseRepository.java (hunk 3 above)
Comment (aalexeen) [reply stock/src/main/java/saffchen/repository/BaseRepository.java:24]: We are using this custom delete method in another method
```
default void deleteExisted(Long id) {
        checkModification(delete(id), id);
    }
```

**Comment by aalexeen** [comment]
> When testing satellites endpoints, I met some issues on my machine:
> 
> 1. When trying to create a new satellite (POST request) if the name of satellite already exists in database I've got HTTP 500 code error.
I'd like to take the error task on me.
> 2. PUT method requests are not allowed by registry.
Needs more details.
> 3. When I try to update a satellite sending by URL non-existing id, then application creates new satellite in database with no errors. I am not sure that such behaviour is expected.
Yes. That is the logic of the update method.
> 4. When I try to update a satellite and send the satellite name that already exists in database I've got HTTP 500 code error.
I'd like to take the error task on me.
> 5. When trying to delete a satellite with non-existing id, got 422 HTTP code error (maybe it's ok), but got no message from IllegalRequestDataException (like Entity with id={id} not found").
I'd like to take the error task on me.

**Comment by pelmegov** [comment]
@aalexeen do you mean you need additional jira for that or you can fix it here?

**Comment by aalexeen** [comment]
I think the additional jira for error handler is better.

**Comment by pelmegov** [comment]
@saffchen please take a look at the comments above.

**Comment by aalexeen** [comment]
> > @saffchen please take a look at the comments above.
> 
> https://jrm-team-1.atlassian.net/browse/JT-44?atlOrigin=eyJpIjoiY2NhODg2MWQyYmUwNDVhYmI3MTczZDg0ZWEwMWNkZWMiLCJwIjoiaiJ9

Got it. I've put it in my scope of view.

**Pull request merged**